puppet/hiera/hiera.yaml

```diff
@@ -1,13 +1,21 @@
 ---
 :backends:
-        - yaml
+  - yaml
 :yaml:
-        :datadir: '/usr/local/puppet/hiera/'
+  :datadir: '/usr/local/puppet/hiera/'
 :hierarchy:
-        - "nodes/external/secret/%{::fqdn}"
-        - "nodes/external/%{::fqdn}"
-        - "nodes/vagrant-%{::vagrant_extra_prefix}%{::is_vagrant}"
-        - "nodes/common"
-        - "vagrant-%{::is_vagrant}"
-        - "common"
-:merge_behavior: deeper
+  # things in nodes/ are part of event-specific repositories, not included in this repo
+  - "nodes/external/secret/%{::fqdn}"
+  - "nodes/external/secret/event-%{::event_name}"
+  - "nodes/external/secret/%{::environment}"
+  - "nodes/external/%{::fqdn}"
+  - "nodes/vagrant-%{::event_name}-%{::is_vagrant}"
+  - "nodes/vagrant-%{::is_vagrant}"
+  - "nodes/event-%{::event_name}"
+  - "nodes/%{::environment}"
+  - "nodes/common"
+  # things below nodes are included in this repo and visible to any event running uber
+  - "vagrant-%{::is_vagrant}"
+  - "%{::environment}"
+  - "common"
+# :merge_behavior: deeper # used to need this, probably not anymore.
```

Comment:
ok one other thing to actually answer your original question:

the files don't override each other.  individual settings in the files override each other.

if the hierarchy was:
common.yaml
production.yaml
event-classic.yaml

and you had the following stuff in each

common.yaml
```
dbhost='localhost'

send_emails='False'
dev_box='True'

1day_badge_price='30'
```

production.yaml
```
send_emails='True'
dev_box='False'
```

event-classic.yaml
```
1day_badge_price='50'
```

then what you end up with at the end is a merging of all those settings, like this:
```
dbhost='localhost'
send_emails='True'
dev_box='False'
1day_badge_price='50'
```

The take-away here is each level of the hierarchy only has to define what it needs / what it needs to override.